### PR TITLE
Add support for the CC1350 LP and Sensortag

### PR DESCRIPTION
**Pull request by g-oikonomou**
This pull adds support for two additional CC13xx boards:

* CC1350LP
* CC1350STK

Sits on top of #1931

### CODE REVIEW

**Comment by aparvizi** [comment]
HI,
I Test CC1350LP_US, cc26XX webdemo with 6lbr :

(on BBB & raspberry pi)

in IEEE mode : 

1) 6lbr up , no problem.
2) web_demo cc1350LP_US connected to 6lbr and get IP's , but no PING .

(6lbr router on BBB & RPI with same config work FINE with CC2650 & ST & CC2358+PA)

 on prob mode:
1) 6lbr up , no problem.
2) web_demo cc1350LP_US not connected & stay in blink, not works;

regards,
a.Parvizi

File: cpu/cc26xx-cc13xx/rf-core/prop-mode.c [hunk 1]
```diff
@@ -364,12 +365,20 @@ prop_div_radio_setup(void)
   uint32_t cmd_status;
   rfc_radioOp_t *cmd = (rfc_radioOp_t *)&smartrf_settings_cmd_prop_radio_div_setup;
 
+  rf_switch_select_path(RF_SWITCH_PATH_SUBGHZ);
```
Comment (tussock-cam) [inline cpu/cc26xx-cc13xx/rf-core/prop-mode.c]: Do we also need to set this in `soft_on_prop`? `rf-ble` calls `rf_core_primary_mode_restore` when it's done, and presumably we want that to switch the antenna over?

**Comment by tussock-cam** [comment]
Ignore my previous comment, I shouldn't have just looked at the diff :)

rf-switch stuff works well on my custom board, haven't tested on launchpad.

**Comment by g-oikonomou** [comment]
Bah you removed it before I could reply! :)

**Comment by g-oikonomou** [comment]
Tell us about your board. Does it have a switch too? Is it 1350-based?

**Comment by tussock-cam** [comment]
It's cc1350 based but has a different switch than the launchpad. We don't have power control over it, so just select/deselect. This setup works nicely for that, though, and lets us enable high impedence when we don't care, which saves a bit of power! Much nicer than my quick little patch to set it high/low when needed.

**Comment by jesse-tussock** [comment]
@g-oikonomou Just following on from @tussock-cam We are using the CC1350 with a SKY13354-368LF switch. It is currently only in prototype. We have made 30 units and are using them in a number of proof of concept projects.

**Comment by g-oikonomou** [comment]
Hi folks, there is new stuff for you to test here.

**Comment by tussock-cam** [comment]
I haven't had a chance to test the new changes, but figure this is probably an okay spot to post this..

`ccxxwares` produces a heap of warnings about unused functions which makes it difficult to find useful warnings in the output. We can silence these by using `-isystem $(TI_XXWARE)` etc. in `Makefile.cc26xx-cc13xx`. I've never used `-isystem` before so not too sure what goes on behind the scenes, but it does seem to change the order things happen in. Comments in the Makefile indicate that we need these included first, so I'm not sure if this would affect that.

If `-isystem` works okay, I think this would be a useful change to improve signal-to-noise :)

**Comment by g-oikonomou** [comment]
I'm not sure what warnings you are referring to. Both my local installation as well as Travis agree that this build is warning-free.

**Comment by aparvizi** [comment]
hi,
i test cc1350LP-US, with RPL_BORDER_ROUTER and cc26xx-web-demo, on prob mode, it's ok,

i want test both mode with 6lbr and RPL, and  send reports  ASAP.

**Comment by aparvizi** [comment]
**CC1350LP_us  ( prefix fd00) , it's ok, in both mode with 6LBR and RPL.**
the problem is in rssi;
rssi is -70 with 30cm distance between  2 node.

the rf config is:

#define DOT_15_4G_CONF_FREQUENCY_BAND_ID DOT_15_4G_FREQUENCY_BAND_915
#define RF_CORE_CONF_CHANNEL                 25

other setting it's default setting.

**Comment by g-oikonomou** [comment]
That's too low :( Both devices US Launchpads?

**Comment by aparvizi** [comment]
yes,

**Comment by aparvizi** [comment]
with TI EASYLINK, the rssi is -20.

**Comment by jonnteolsson** [comment]
All looks good for CC13xx, tested on CC1310LaunchPad, CC1350LaunchPad and CC1350SensorTag.

**Comment by stringr9** [comment]
Works great with the CC1310 LP.  I've seen the same issue as @aparvizi with the CC1350_EU LP.  Boards 30cm apart shows -70dBm.  Switching one with a CC1310LP increases RSSI to -45dBm.  With two CC1310 LP the RSSIs are around -20dBm.

**Comment by g-oikonomou** [comment]
@aparvizi @stringr9 we think we may be onto something that could be causing the RSSI problems with the CC1350LP. I pushed a couple of commits and planning to test first chance. Can you please test too?

**Comment by stringr9** [comment]
No change on my end with the 1350 LP unfortunately.  Still seeing ~-70dBm when 30cm apart.

**Comment by aparvizi** [comment]
also, no change.

**Comment by g-oikonomou** [comment]
OK well we're suspecting the RF switch manipulation here...

**Comment by g-oikonomou** [comment]
More commits added to make overrides even more configurable. Most importantly: I just realised that TX power values used in `CMD_PROP_DIV_RADIO_SETUP` differ across boards, so I've reworked the driver to allow the board code to provide correct values. Fallback to defaults possible.

I am seeing very very sane RSSI values here using CC1310 (LP and EM) and CC1350 (LP and STK) devices in the same testbed. STK values are somewhat lower.

@aparvizi @stringr9 @jonnteolsson @tussock-cam can you confirm please?

Fixes #1659

**Comment by g-oikonomou** [comment]
Shamelessly also pinging @Johan-Henry, @arurke and @jimmycmlo

**Comment by arurke** [comment]
Do you expect any changes to cc1310em? I do not have any cc1350 LP/STK.

**Comment by g-oikonomou** [comment]
For EM the RSSI values should be comparable to those observed when using SmartRF studio

**Comment by aparvizi** [comment]
Dear g-oikonomou ,

I test on cc1350-lp-US , in 6LBR, both side is cc1350-lp-US,
worked fine, with stable ping.
i will send full report after finish all test.
Thank you.

**Comment by aparvizi** [comment]
in prob mode with 30 cm distance:
rssi:-17

in 30 meter distance with 3 Walls:
rssi:-54

(ping time for both is attached):

cc-1350 30cm distance
cc1350-us 30m with 3 walls
ping_prob_mode


in IEEE mode:
in 30 meter distance with 3 Walls:
rssi:-80
(but, ping time is good)-all attached:

ieee_mod_30meter
ieee_ping

**Comment by jonnteolsson** [comment]
Finally it looks good on all targets:
BR cc1350lp <-> cc1350lp, -14dBm in both directions
BR cc1350lp <-> cc1350stk, -25dBm on STK, -16dBm on LP
BR cc1350lp <-> cc1310lp, -14dBm on cc1310lp, -14dBm on LP

**Comment by stringr9** [comment]
Testing RSSIs on CC1310/50 LPs look good!  Similar values to @jonnteolsson above.


Unfortunately I'm encountering some kind of issue with contikimac on this branch that's severely limiting range.   Testing using CC1310 LPs.  Findings summarised in this table:


|  Commit      | RDC |  Range for PER > 20%           | RSSI  |
| ------------- | ------- |:-------------:| -----|
| a6b14a0 | contikimac  | 20m | -50 |
| a6b14a0 | nullrdc  | 260m      |   -95 |
| edec34f | contikimac  | 140m     |    -93 |
| edec34f | nullrdc  | 265m      |    -93 |

I conducted the tests with a simple modification of the simple-udp-rpl examples to calculate PER of received packets.  Can be found [here](https://github.com/stringr9/contiki/tree/master/examples/ipv6/per-test).  For the problematic test at 20m, the RSSI of the rx'd packets that did get through seemed fine (~-50) but only a fraction of the packets were getting through to the receiver.  At 30m no packets were received at all.


Sounds like @aparvizi was able to get 30m  without any issues so I'm hoping this is an oversight on my part, but I ran the exact same example on both commits with no other changes so it's hard to see how.

**Comment by EduardoMolinaTomas** [comment]
First I apologize you in case this wouldn't were te correct place to ask this question. 

I've working with this branch since is the only one which works with CC1350 correctly. In my case y only have CC1350 Launchpads as development kits due CC1350 is not available yet to my country.

Despite of this, for my project requirements I had to made a custom board based on CC1350 sensortag whitout all sensors included in order to reduce the board size to minimum. The schematic is the same.  

This custom boards seem to comunicate correctly with Launchpads using Smart RF Studio from Texas Instrumens. Also I compiled some test codes from code composer and both sub 1Ghz and BLE seems to be ok.

Unfortunately I can't make it work under Contiki.

The only changes I've made was to comment some initializations related with I2C in "board.c" files from "...\contiki\platform\srf06-cc26xx\sensortag" as I don't have this peripherals connected. By this way I'm able to run some test codes, for instance a simple led blinking.

I've tried to cc26xx-web-demo and a modified code which only acts as 6lbr client whitout success. However, these codes compiled and tested on their counterparts CC1350 Launchpads are working right.

I wonder if it would be possible to have any advice of what I'm doing wrong.

Besides I'm trying Rime examples and I found interesting due its simplicity. Where I could get more information about this topic and Contiki in general?

Thank you in advance for your time.

**Comment by rifotu** [comment]
Hello,

I tried this pull request with the below setup

2x CC1350 ST  with web-demo firmware  (nullrdc driver setting in project-conf.h)
2x CC1310 LP  with web-demo firmware (default driver) latest contiki master branch commit: cf2b2497..)
 
6LBR (1.4.1) with a CC1310 Launchpad slip radio (latest contiki master branch commit: cf2b24971...)

The systems works but I faced some problems, I would be happy if you can comment on this.

1. Range of CC1350 ST's are problematic. Can't take them further than 6 meters from Slip Radio

2. When moving around CC1350 ST, sometimes they lose connection to 6LBR and can't reestablish the connection no matter how close I place them to 6LBR. I have to do a manual restart to solve this.

**Comment by jesse-tussock** [comment]
@g-oikonomou The latest code here fixes our synth calibration issues. Presumably related to the tx power refactor. Seems to be working well now.

**Comment by g-oikonomou** [comment]
I'll merge this in a month unless I receive an objection.

**Pull request merged**